internal/service/rds/cluster_activity_stream_test.go

```diff
@@ -0,0 +1,227 @@
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSClusterActivityStreamConfig(clusterName, instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRDSClusterActivityStreamExists(resourceName, &dbCluster),
+					testAccCheckAWSRDSClusterActivityStreamAttributes(&dbCluster),
+					acctest.MatchResourceAttrRegionalARN(resourceName, "resource_arn", "rds", regexp.MustCompile("cluster:"+clusterName)),
+					resource.TestCheckResourceAttrPair(resourceName, "resource_arn", rdsClusterResourceName, "arn"),
+					resource.TestCheckResourceAttrPair(resourceName, "kms_key_id", kmsKeyResourceName, "key_id"),
+					resource.TestCheckResourceAttrSet(resourceName, "kinesis_stream_name"),
+					resource.TestCheckResourceAttr(resourceName, "mode", rds.ActivityStreamModeAsync),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+}
+
+func TestAccAWSRDSClusterActivityStream_disappears(t *testing.T) {
+	var dbCluster rds.DBCluster
+	clusterName := sdkacctest.RandomWithPrefix("tf-testacc-aurora-cluster")
+	instanceName := sdkacctest.RandomWithPrefix("tf-testacc-aurora-instance")
+	resourceName := "aws_rds_cluster_activity_stream.test"
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:     func() { acctest.PreCheck(t) },
+		ErrorCheck:   acctest.ErrorCheck(t, rds.EndpointsID),
+		Providers:    acctest.Providers,
+		CheckDestroy: testAccCheckAWSClusterActivityStreamDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSClusterActivityStreamConfig(clusterName, instanceName),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSRDSClusterActivityStreamExists(resourceName, &dbCluster),
+					acctest.CheckResourceDisappears(acctest.Provider, tfrds.ResourceClusterActivityStream(), resourceName),
+				),
+				ExpectNonEmptyPlan: true,
+			},
+		},
+	})
+}
+
+func testAccCheckAWSRDSClusterActivityStreamExists(resourceName string, dbCluster *rds.DBCluster) resource.TestCheckFunc {
+	return testAccCheckAWSRDSClusterActivityStreamExistsWithProvider(resourceName, dbCluster, acctest.Provider)
+}
+
+func testAccCheckAWSRDSClusterActivityStreamExistsWithProvider(resourceName string, dbCluster *rds.DBCluster, provider *schema.Provider) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("not found: %s", resourceName)
+		}
+
+		if rs.Primary.ID == "" {
+			return fmt.Errorf("DBCluster ID is not set")
+		}
+
+		conn := provider.Meta().(*conns.AWSClient).RDSConn
+
+		response, err := tfrds.FindDBClusterByClusterArn(conn, rs.Primary.ID)
+
+		if err != nil {
+			if tfresource.NotFound(err) {
+				return fmt.Errorf("DBCluster not found")
+			}
+			return err
+		}
+
+		dbCluster = response
+
+		return nil
+	}
+}
+
+func testAccCheckAWSRDSClusterActivityStreamAttributes(v *rds.DBCluster) resource.TestCheckFunc {
```

Comment:
why is these need if attributes are already checked?

Reply:
It checks properties returned from AWS and not only attributes stored with the resource. i.e. the `ActivityStreamStatus` which is expected to be in state `Started`